src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,300 @@
+                }
+            ],
+            "TargetCapacitySpecification": {
+                "TotalTargetCapacity": count,
+                "DefaultTargetCapacityType": "on-demand"
+                if self._queue_config["CapacityType"] == "ONDEMAND"
+                else "spot",
+            },
+            "Type": "instant",
+            "SpotOptions": {
+                "AllocationStrategy": self._queue_config["AllocationStrategy"],
+                "SingleInstanceType": False,
+                "SingleAvailabilityZone": True,  # Set to False for Multi-AZ support
+                # If the minimum target capacity is not reached, the fleet launches no instances
+                "MinTargetCapacity": 1 if not self._all_or_nothing else count,
+            },
+            "OnDemandOptions": {
+                "AllocationStrategy": self._queue_config["AllocationStrategy"],
+                "CapacityReservationOptions": {"UsageStrategy": "use-capacity-reservations-first"},
+                "SingleInstanceType": False,
+                "SingleAvailabilityZone": True,  # Set to False for Multi-AZ support
+                # If the minimum target capacity is not reached, the fleet launches no instances
+                "MinTargetCapacity": 1 if not self._all_or_nothing else count,
+            },
+        }
+
+        launch_params.update(launch_overrides)
+        if launch_overrides:
+            logger.info("Found CreateFleet parameters override. Launching instances with: %s", launch_params)
+        return launch_params
+
+    def _launch_instances(self, launch_params):
+        """Launch a batch of ec2 instances."""
+        try:
+            response = create_fleet(self._region, self._boto3_config, launch_params)
+            for error in response.get("Errors", []):
+                logger.error("Error in CreateFleet request: %s - %s", error.get("ErrorCode"), error.get("ErrorMessage"))
+
+            logger.debug("CreateFleet response: %s", response)
+            instance_ids = [inst_id for instances in response["Instances"] for inst_id in instances["InstanceIds"]]
+            return {"Instances": self._get_instances_info(instance_ids)}
+        except ClientError as e:
+            logger.error("Failed CreateFleet request: %s", e.response.get("ResponseMetadata").get("RequestId"))
+            raise e
+
+    def _get_instances_info(self, instance_ids, retry=3):
+        """
+        Describe instances to retrieve info not available from create-fleet response.
+
+        :raises EC2InstancesInfoUnavailable if after N retries it's not possible to retrieve info for all the instances
+        :raises ClientError in case of boto3 failure
+        """
+        # Wait for instances to be available in EC2
+        time.sleep(5)
+
+        # Retrieve instance info from Instance Ids
+        instances = []
+        if instance_ids:
+            logger.info(f"Retrieving instance info for instances: {instance_ids}")
```

Comment:
Shall we move this log to the _launch_instances function rephrasing to something like: launched the following EC2 instances:?

Reply:
done